src/applications/user-testing/disability-benefits-all-claims-conditions-chapter/content/secondaryEnhanced.jsx

```diff
@@ -42,24 +42,43 @@ export const SecondaryEnhancedOptionsConflictingAlert = () => (
   </va-alert>
 );
 
-export const createSecondaryEnhancedDescriptionString = causedByCondition => {
-  const conditions = Object.keys(causedByCondition || {}).filter(
-    key => causedByCondition[key],
+export const createSecondaryEnhancedDescriptionString = (
+  causedByConditionObj,
+  fullData = {},
+) => {
+  const { conditions = [], ratedDisabilities = [] } = fullData;
+
+  // names that STILL exist
+  const validNames = new Set([
+    // new conditions that have NOT been deleted
+    ...conditions
```

Comment:
I know our F/E isn't typed....but is there a link somewhere of what these objects will look like?

Reply:
IE, can `name` ever be `undefined`?

Reply:
This change only lives in the Papaya enhanced workflow prototype, which may be dropped altogether. For now, the conditions and ratedDisabilities are objects the form system injects into formData during local testing. The 526 JSON schema lists name as required string, so it should never be undefined. Hence, the optional chaining guard offers a lightweight safety net in case it is.